tmva/sofie/test/TestGemmDerivative.cxx

```diff
@@ -0,0 +1,115 @@
+#include <TInterpreter.h>
+#include <TInterpreterValue.h>
+
+#include "gtest/gtest.h"
+
+#include <stdexcept>
+
+class GemmTest : public testing::TestWithParam<std::tuple<int, int, int>> {
+public:
+   void SetUp() override
+   {
+      m = std::get<0>(GetParam());
+      n = std::get<1>(GetParam());
+      k = std::get<2>(GetParam());
+   }
+
+protected:
+   int m;
+   int n;
+   int k;
+};
+
+void validator(float *variables, int m, int n, int k, float *target)
+{
+   for (int i = 0; i < m * k + k * n + 1; ++i) {
+      target[i] = 0;
+      if (i == 0) {
+         // Check derivative with alpha
+         int _a;
+         int _b;
+         for (int _m = 1; _m < m + 1; ++_m) {
+            for (int _n = 1; _n < n + 1; ++_n) {
+               for (int _k = 1; _k < k + 1; ++_k) {
+                  _a = (_k - 1) * m + (_m - 1) % m + 1;
+                  _b = (_n - 1) * k + (_k - 1) % k + 1;
+                  target[i] += variables[_a] * variables[_b + m * k];
+               }
+            }
+         }
+      } else if ((i - 1) < m * k) {
```

Comment:
```suggestion
   // Check derivative with alpha
   for (int _m = 1; _m < m + 1; ++_m) {
      for (int _n = 1; _n < n + 1; ++_n) {
         for (int _k = 1; _k < k + 1; ++_k) {
            int _a = (_k - 1) * m + (_m - 1) % m + 1;
            int _b = (_n - 1) * k + (_k - 1) % k + 1;
            target[0] += variables[_a] * variables[_b + m * k];
         }
      }
   }

   for (int i = 1; i < m * k + k * n + 1; ++i) {
      if ((i - 1) < m * k) {
```
Personally, I would avoid some indentation here and make the code more readable by putting the `alpha` section outside the loop.